packages/snaps-utils/src/namespace.ts

```diff
@@ -85,11 +87,14 @@ export const LimitedString = size(string(), 1, 40);
  */
 export const ChainIdStruct = pattern(string(), CHAIN_ID_REGEX);
 export type ChainId = `${string}:${string}`;
+export type Caip2ChainId = Infer<typeof ChainIdStruct>;
```

Comment:
This should be the same as `ChainId`? Although better typed

Reply:
I'm willing to change `Caip2ChainId` to `ChainId` and get rid of the original `ChainId`,  I do want the better typing. We also re-export `ChainId` in `snaps-types`, so I'd like it to be accurate against CAIP-2 when exposing to end users.

Reply:
Turns out it is not better typed, it is in fact worse, it becomes a string. There was an extension engineer that observed this when they were porting over the types to `@metamask/utils`.